src/components/shared/NodeCountSelector.js

```diff
@@ -16,15 +16,17 @@ const DEFAULT_VALUE_CONSTRAINTS = {
  * max values for scaling limits.
  */
 class NodeCountSelector extends React.Component {
-  handleFormSubmit = (e) => {
-    e.preventDefault();
-  };
-
   updateValue(valuesToAdd) {
-    const nextScalingValue = Object.assign({}, this.props.scaling, valuesToAdd);
+    setTimeout(() => {
```

Comment:
Forcing the callback to be executed after `react` makes all the updates.

This is because the 2 underlying `<NumberPicker />` components execute the callback when their `min`/`max` props change (because they also need to validate again). This also includes the first time that the `min`/`max` props are set, which is when the components get mounted.

Because the 2 callbacks (for 2 `<NumberPicker />` components) get executed one right after another, the value of the scaling in the props does not have time update in between calls. Therefore the callback returns that the input is invalid when it's actually valid.

Long story short, this makes sure that the scaling is valid by default and it doesn't prevent node pool creation.